server/routes/events.js

```diff
@@ -2,14 +2,262 @@
+                return next(err);
+            }
+            res.send(result.rows);
+        });
+    } else if (company != undefined) {
+        db.query('SELECT event_id FROM event_company WHERE company_id = \'' + company + '\'', [], (err, result) => {
+            if (err) {
+                return next(err);
+            }
+            res.send(result.rows);
+        });
+    } else if (featured != undefined) {
+        db.query('SELECT * FROM event WHERE is_featured = \'' + featured + '\'', [], (err, result) => {
+            if (err) {
+                return next(err);
+            }
+            res.send(result.rows);
+        });
+    } else if (past != undefined) {
+        db.query('SELECT fb_id FROM event WHERE ends_at < now()', [], (err, result) => {
+            if (err) {
+                return next(err);
+            }
+            res.send(result.rows);
+        });
+    } else {
+        res.json({result: false}); // error message
+    }
 });
 
-// GET events listing. 
+// GET all events
 router.get('/', function(req, res, next) {
-    // TODO
+  db.query('SELECT * FROM event', [], (err, result) => {
+    if (err) {
+      return next(err);
+    }
+    res.send(result.rows);
+  });
+});
+
+// GET all event names
+router.get('/names', function(req, res, next) {
+  db.query('SELECT name FROM event', [], (err, result) => {
+    if (err) {
+      return next(err);
+    }
+    res.send(result.rows);
+  });
+});
+
+// GET all events containing term substring
+// I think this needs to go before all of the /:event_id, or else it interprets search as an event id :/
+// also I can't figure out how to use the [] for this one :(
+router.get('/search', function(req, res, next) {
+  const term = req.query.term;
+	
+  db.query('SELECT * FROM event WHERE name ILIKE \'%' + term + '%\'', [], (err, result) => {
+    if (err) {
+      return next(err);
+    }
+    res.send(result.rows);
+  });
+});
+
+// GET all unique event locations
+router.get('/locations', function(req, res, next) {
+  db.query('SELECT DISTINCT location_id FROM event', [], (err, result) => {
+  	if (err) {
+      return next(err);
+  	}
+  	res.send(result.rows);
+  });
+});
+
+// GET event by event_id
+router.get('/:event_id', function(req, res, next) {
+  const event_id = req.params.event_id;
+  db.query('SELECT * FROM event WHERE fb_id = $1', [event_id], (err, result) => {
+	if (err) {
+	  return next(err);
+	}
+	res.send(result.rows);
+  });
+});
+
+// GET all users that have favorited the given event
+router.get('/:event_id/favorites', function(req, res, next) {
+  const event_id = req.params.event_id;
+  db.query('SELECT user_id FROM favorite_events WHERE event_id = $1', [event_id], (err, result) => {
+	if (err) {
+	  return next(err);
+	}
+	res.send(result.rows);
+  });
+});
+
+// GET all users registered to a given event
+router.get('/:event_id/register', function(req, res, next) {
+  const event_id = req.params.event_id;
+  const paid = req.query.paid;
```

Comment:
You need to convert this parameter to boolean type explicitly. If the user gives a query like `http://localhost:5000/events/1343475239125573/register?paid=qw` you're going to get an error. So, if you can successfully convert to bool then use paid var, otherwise do same action as if paid was undefined.